lib/controllers/game_controller.cpp

```diff
@@ -435,6 +449,23 @@ void GameController::loadEndGameLevel() {
     scene_manager->createScene<EndScene>(*entityFactory, *engine);
 }
 
+void GameController::pauseGame() {
+    if(having_a_break){
+        scene_manager->destroyScene(PauseScene::getLayerStatic());
+        game_state_manager->setState(GameState::InGame);
+        having_a_break = false;
+    }else{
+        // Checks whether an intermission to a new level is going on. When this is the case,
+        // the break won't be initiated and the player has to wait for the next level.
+        if(!scene_manager->isSceneActive<IntermissionScene>()){
+            scene_manager->destroyScene(PauseScene::getLayerStatic());
+            scene_manager->createScene<PauseScene>(*entityFactory, *engine, [this](){this->pauseGame();},
+                                                    [this](){this->loadMainMenu();});
+            having_a_break = true;
+        }
+    }
+}
```

Comment:
Isn't `having_a_break` the same as checking whether the PauseScene is currently active?

I don't really like the hard coded way of checking whether the intermission scene is active, you shouldn't have to specify a specific scene here. And you should be able to pause during the intermission, as it really is still in game and you should be able to pause.